### ISSUE TITLE
Feature Request: Allow functions in Flutter to be converted into stateless widgets

### ISSUE THREAD

**Issue by itsahmed-dev** [Dart-Code/Dart-Code#4513]
VS Code Version: 1.77.3
OS: Windows_NT x64 10.0.22000
Extensions: Dart & Flutter: 3.62.0

Code action to convert a function that returns a widget to a stateless widget. A stateful widget can be converted into stateless and vice versa but it should be possible to convert a function that returns a widget into a stateless widget as well.

![Code Action for a Widget](https://user-images.githubusercontent.com/94858768/234894186-54d9bfac-f40b-4c7e-915e-5c12585edf81.png)

![Code Action for a Function](https://user-images.githubusercontent.com/94858768/234894263-6fd8530f-a68b-4ba3-a69f-4869b2435beb.png)

**Here is why I'm asking for it**

[How to improve the performance of your Flutter app](https://blog.codemagic.io/how-to-improve-the-performance-of-your-flutter-app./)

[Why should I ever use stateless widgets instead of functional widgets?](https://github.com/flutter/flutter/issues/19269)

[Methods vs Stateless Widget to return widgets](https://www.reddit.com/r/FlutterDev/comments/mp4xpx/methods_vs_stateless_widget_to_return_widgets/)

```
Widget dots({required PageController controller, required bool isDarkMode}) {
  return SmoothPageIndicator(
    controller: controller,
    count: 5,
    onDotClicked: (index) => controller.animateToPage(
      index,
      duration: const Duration(milliseconds: 200),
      curve: Curves.linear,
    ),
    effect: JumpingDotEffect(
      dotHeight: 6.0,
      dotWidth: 6.0,
      activeDotColor: isDarkMode ? Colors.white : kPrimaryColor900,
      dotColor: isDarkMode ? Colors.grey : kPrimaryColor100,
    ),
  );
}
```

**Comment by DanTup** [comment]
@itsahmed-dev sorry for the delay, I missed the original notification about this. If you'd still like to see this, it would be best raised at https://github.com/dart-lang/sdk which is where the Dart analysis server lives (which provides these actions to VS Code).

Thanks!

**Comment by itsahmed-dev** [comment]
> @itsahmed-dev sorry for the delay, I missed the original notification about this. If you'd still like to see this, it would be best raised at https://github.com/dart-lang/sdk which is where the Dart analysis server lives (which provides these actions to VS Code).
> 
> Thanks!

Yeah I figured that would be the case so I've done it already. Thanks
